XTS/localjunit/crash-recovery-tests/src/main/java/com/arjuna/qa/extension/JBossAS7ServerKillProcessor.java

```diff
@@ -16,15 +16,15 @@ public class JBossAS7ServerKillProcessor extends BaseServerKillProcessor {
     private static final String PROCESSES_CMD = ServerExtension.OSType.getOSType().getPSCommand();
     private static final String PS_AUX_CMD = ServerExtension.isSolaris() ? "/usr/ucb/ps aux" : "ps aux";
     private static final String CHECK_JBOSS_ALIVE_CMD = "if [ \"x`" + PROCESSES_CMD + " | grep 'jboss-module[s]'`\" = \"x\" ]; then exit 1; fi";
-    private static final String SHUTDOWN_JBOSS_CMD = PROCESSES_CMD + " | grep jboss-module[s] | awk '" + ServerExtension.OSType.getOSType().getPSIDIndex() + "' | xargs kill";
-    private static final String CHECK_FOR_DEFUNCT_JAVA_CMD = "if [ \"x`" + PS_AUX_CMD + " | grep '\\[java\\] <defunct>'`\" = \"x\" ]; then exit 1; fi";  
-    
+    private static final String SHUTDOWN_JBOSS_CMD = "pgrep -f 'jboss-module[s]' | xargs kill";
```

Comment:
FYI this is also solving the cases when the USER column  of the 'ps aux' command is empty.